packages/std/src/sections.rs

```diff
@@ -6,6 +9,33 @@ pub fn decode_sections2(data: Vec<u8>) -> (Vec<u8>, Vec<u8>) {
     (first, second)
 }
 
+/// Encodes multiple sections of data into one vector.
+///
+/// Each section is suffixed by a section length encoded as big endian uint32.
+/// Using suffixes instead of prefixes allows reading sections in reverse order,
+/// such that the first element does not need to be re-allocated if the contract's
+/// data structure supports truncation (such as a Rust vector).
+///
+/// The resulting data looks like this:
+///
+/// ```ignore
+/// section1 || section1_len || section2 || section2_len || section3 || section3_len || …
+/// ```
+#[allow(dead_code)] // used in Wasm and tests only
+pub fn encode_sections(sections: &[Vec<u8>]) -> StdResult<Vec<u8>> {
+    let mut out_len: usize = sections.iter().map(|section| section.len()).sum();
+    out_len += 4 * sections.len();
+    let mut out_data = Vec::with_capacity(out_len);
+    for section in sections {
+        let section_len = to_u32(section.len())?.to_be_bytes();
```

Comment:
You can use assertions here and avoid all the error handling because in Wasm the contract can never create a vector with a capacity > 4 GB. This means each input section is smaller than 4 GB and `section.len()` fits in u32.

Reply:
I found an even simpler explanation: usize = u32 on the platform we care about. See https://github.com/CosmWasm/cosmwasm/pull/792 for an approach to express this in code.

Reply:
Of course. Thanks for this, it makes the code simpler.